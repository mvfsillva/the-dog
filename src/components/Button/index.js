import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { transitions } from 'polished'
import { transition } from '@mixins/transition'

const fontSize = ({ height }) => `${height / 40}rem`

const StyledButton = styled.button`
  display: inline-flex;
  align-items: center;
  white-space: nowrap;
  padding: 15px 50px;
  border-radius: 0.125em;
  box-sizing: border-box;
  cursor: pointer;
  background-color: ${props => (props.outlined ? 'transparent' : props.theme.colors.gray.default)};
  color: ${props => (props.outlined ? props.theme.colors.gray.default : props.theme.colors.white)};
  font-size: ${fontSize};
  font-family: ${props => props.theme.fonts.primary};
  letter-spacing: 3px;
  line-height: 20px;
  ${transitions(transition({ property: 'background-color', duration: '250ms' }))};
  ${transitions(transition({ property: 'border-color', duration: '250ms' }))};

  &:hover,
  &:focus,
  &:active {
    background-color: ${props =>
      props.outlined ? props.theme.colors.gray.default : 'transparent'};
    color: ${props =>
      props.outlined ? props.theme.colors.white : props.theme.colors.gray.default};
  }

  &:focus {
    outline: none;
  }
`

const Button = ({ type, ...props }) => <StyledButton type={type} {...props} />

Button.propTypes = {
  outlined: PropTypes.bool,
  height: PropTypes.number,
  type: PropTypes.string,
}

Button.defaultProps = {
  type: 'button',
  height: 40,
}

export default Button
