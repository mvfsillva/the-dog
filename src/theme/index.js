const baselineGrid = 4
const interfaceGrid = 8

const breakpoints = ['576px', '768px', '992px', '1200px']

const colors = {
  primary: {
    default: '#6800c3',
    fallback: '#a55cff',
    text: '#fff',
  },
}

export default {
  baselineGrid,
  interfaceGrid,
  breakpoints,
  colors,
}
