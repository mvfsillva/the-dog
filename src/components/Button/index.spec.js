import React from 'react'
import 'jest-styled-components'
import theme from '../../theme'
import { shallowWithTheme, mountWithTheme } from '../../../config/test/enzyme.theme'
import Button from '.'

const shallow = shallowWithTheme(theme)
const mount = mountWithTheme(theme)

const wrap = (props = {}) => shallow(<Button {...props} />).dive()

describe('[Component: Button]', () => {
  it('mounts with different combination props', () => {
    mount(<Button>Teste</Button>)
    mount(<Button outlined>Teste</Button>)
  })

  it('renders props when passed in', () => {
    const wrapper = wrap({ type: 'submit' })
    expect(wrapper.find({ type: 'submit' })).toHaveLength(1)
  })

  it('renders button by default', () => {
    const wrapper = wrap()
    expect(wrapper.find('button')).toHaveLength(1)
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<Button>Button component</Button>)
    expect(wrapper).toMatchSnapshot()
  })
})
