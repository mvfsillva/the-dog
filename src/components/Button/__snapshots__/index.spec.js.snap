// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: Button] matches snapshot 1`] = `
<styled.button
  height={40}
  type="button"
>
  Button component
</styled.button>
`;
